data-processing-lib/python/src/data_processing/transform/pipelined_transform.py

```diff
@@ -0,0 +1,202 @@
+        tables = self._get_table_list(r_bytes)
+        return tables, r_metadata
+
+    def _get_table_list(self, transformed: list[tuple[pa.Table, str]]):
+        tables = []
+        for t in transformed:
+            tables.append(t[0])
+        return tables
+
+    def _apply_transforms_to_data(
+        self, transforms: list[AbstractBinaryTransform], data: list[tuple[str, bytearray]]
+    ) -> tuple[list[tuple[bytes, str]], dict[str, Any]]:
+        r_bytes = []
+        r_metadata = {}
+        for datum in data:
+            processed, metadata = self._apply_transforms_to_datum(transforms, data)
+            r_bytes.extend(processed)
+            r_metadata = r_metadata | metadata
+
+        return r_bytes, r_metadata
+
+    def _apply_transforms_to_datum(
+        self,
+        transforms: Union[list[AbstractBinaryTransform], list[AbstractTableTransform]],
+        datum: Union[tuple[str, bytearray], tuple[str, pa.Table]],
+    ) -> tuple[list[tuple[bytes, str]], dict[str, Any]]:
+        """
+        Apply the list of transforms to the given datum tuple of filename and byte[]
+        Args:
+            transforms:
+            datum:
+        Returns: same as transform_binary().
+
+        """
+        r_metadata = {}
+        pending_to_process = [(datum[0], datum[1])]
+        for transform in transforms:
+            transform_name = type(transform).__name__
+            to_process = pending_to_process
+            pending_to_process = []
+            for tp in to_process:  # Over all outputs from the last transform  (or the initial input)
+                fname = tp[0]
+                to_transform = tp[1]
+                transformation_tuples, metadata = self._call_transform(transform, to_transform, fname)
+                # Capture the list of outputs from this transform as inputs to the next (or as the return values).
+                for transformation in transformation_tuples:
+                    transformed, extension = transformation
+                    fname = transform_name + "-output" + extension
+                    next = (fname, transformed)
+                    pending_to_process.append(next)
+                # TODO: this is not quite right and might overwrite previous values.
+                # Would be better if we could somehow support lists.
+                r_metadata = r_metadata | metadata
+
+        # Strip the pseudo-base filename from the pending_to_process tuples, leaving only the extension, as required.
+        r_bytes = []
+        for tp in pending_to_process:
+            fname = tp[0]
+            byte_array = tp[1]
+            extension = pathlib.Path(fname).suffix
+            tp = (byte_array, extension)
+            r_bytes.append(tp)
+
+        return r_bytes, r_metadata
+
+    def _call_transform(
+        self,
+        transform: Union[AbstractTableTransform, AbstractBinaryTransform],
+        datum: Union[pa.Table, bytearray],
+        file_name: str,
+    ):
```

Comment:
This seems to be wrong. Transform is always binary. A table is just one of the cases of binary. no need for Unions. Take a look at https://github.com/IBM/data-prep-kit/blob/dev/data-processing-lib/python/src/data_processing/transform/table_transform.py. it takes binary and returns binary. Table is just an intermediate format